### PR TITLE
Remove Lua governance and Lua dependency

### PR DESCRIPTION
**Pull request by eddyashton**
Removes Lua governance, and our Lua interpreter, and `3rdparty/internal/lua`, and a bunch of associated now-dead code.

I've fixed `proposal_id_test` to use JS governance, but not `member_voting_test`. I've simply commented that out for now so we can work out at another point what's worth saving+porting and what can be binned.

I've done a basic pass on the docs replacing the sample proposals and votes (and responses) with their JS equivalents, but haven't tackled the text yet.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Removed
 
 - `/gov/query` and `/gov/read` governance endpoints are removed (#2442).
+- Lua governance is removed. `JS_GOVERNANCE` env var is no longer necessary, and JS constitution is the only governance script which must be provided and will be used by the service.
```
Comment (achamayou) [inline CHANGELOG.md:22]:
```suggestion
- Lua governance is removed. `JS_GOVERNANCE` env var is no longer necessary, and JS constitution is the only governance script which must be provided and will be used by the service. `--gov-script` can no longer be passed to `cchost`.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -386,56 +379,40 @@ if(BUILD_TESTS)
...
-      PRIVATE ${CMAKE_THREAD_LIBS_INIT} lua.host http_parser.host sss.host
-              ccf_endpoints.host quickjs.host
-    )
-
-    set_tests_properties(
-      member_voting_test
-      PROPERTIES ENVIRONMENT
-                 RUNTIME_CONFIG_DIR=${CMAKE_SOURCE_DIR}/src/runtime_config
-    )
+    # TODO: Port or remove entirely? add_unit_test( member_voting_test
```
Comment (achamayou) [inline CMakeLists.txt:391]: I will have a look at porting what's missing from the end to end tests, unless there is a specific pattern we can't test there.

File: src/http/http_rpc_context.h [hunk 3]
```diff
@@ -269,8 +269,16 @@ namespace http
     virtual void set_response_body(std::string&& body) override
     {
       response_body = std::vector<uint8_t>(body.begin(), body.end());
-      set_response_header(
-        http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
+
+      // If there's no explicit content type already set, assume this is really
```
Comment (achamayou) [inline src/http/http_rpc_context.h:273]: I wonder if setting a default content type is the right thing to do. It seems to me that it's only made bugs slightly harder to detect for us, we would have caught them earlier if there was no default.

Comment (eddyashton) [reply src/http/http_rpc_context.h:273]: I think that's a fine approach too. It changes the responses for any endpoints that were silently benefitting from this, but that's not too many, not too dangerous, and better they find out now? Yes I think I've convinced myself, going to remove the default entirely and see what breaks.

File: src/node/rpc/test/proposal_id_test.cpp [hunk 4]
```diff
@@ -74,14 +80,17 @@ DOCTEST_TEST_CASE("Unique proposal ids")
...
+  // Count retries to confirm that these proposals conflicted and one was
+  // retried (potentially multiple times, if very unlucky and gets a retried
+  // root before the earlier transaction has set it)
   auto metrics_req = create_request(nlohmann::json(), "api/metrics", HTTP_GET);
   auto metrics = frontend_process(frontend, metrics_req, proposer_cert);
   auto metrics_json = serdes::unpack(metrics.body, serdes::Pack::Text);
   for (auto& row : metrics_json["metrics"])
   {
-    if (row["path"] == "proposals")
+    if (row["path"] == "proposals.js")
```
Comment (achamayou) [inline src/node/rpc/test/proposal_id_test.cpp:91]:
```suggestion
    if (row["path"] == "proposals")
```

File: src/node/rpc/test/proposal_id_test.cpp [hunk 5]
```diff
@@ -158,30 +169,24 @@ DOCTEST_TEST_CASE("Compaction conflict")
...
-  proposal.script = std::string(R"xxx(
-    tables, member_info = ...
-    return Calls:call("new_member", member_info)
-  )xxx");
-  proposal.parameter["cert"] = proposed_member;
-  proposal.parameter["encryption_pub_key"] = dummy_enc_pubk;
+  nlohmann::json proposal_body = "Ignored";
   const auto propose =
-    create_signed_request(proposal, "proposals", kp, proposer_cert);
+    create_signed_request(proposal_body, "proposals.js", kp, proposer_cert);
```
Comment (achamayou) [inline src/node/rpc/test/proposal_id_test.cpp:174]:
```suggestion
    create_signed_request(proposal_body, "proposals", kp, proposer_cert);
```

File: src/node/rpc/test/proposal_id_test.cpp [hunk 6]
```diff
@@ -158,30 +169,24 @@ DOCTEST_TEST_CASE("Compaction conflict")
...
+  const auto out = parse_response_body<jsgov::ProposalInfoSummary>(rs);
   DOCTEST_CHECK(out.state == ProposalState::OPEN);
 
   auto metrics_req = create_request(nlohmann::json(), "api/metrics", HTTP_GET);
   auto metrics = frontend_process(frontend, metrics_req, proposer_cert);
   auto metrics_json = serdes::unpack(metrics.body, serdes::Pack::Text);
   for (auto& row : metrics_json["metrics"])
   {
-    if (row["path"] == "proposals")
+    if (row["path"] == "proposals.js")
```
Comment (achamayou) [inline src/node/rpc/test/proposal_id_test.cpp:189]:
```suggestion
    if (row["path"] == "proposals")
```

File: tests/governance_history.py [hunk 7]
```diff
@@ -47,10 +46,8 @@ def count_governance_operations(ledger):
                         cert, sig, req, request_body, digest
                     )
                     request_target_line = req.decode().splitlines()[0]
-                    if "/gov/proposals" in request_target_line:
-                        vote_suffix = (
-                            "/ballots" if os.getenv("JS_GOVERNANCE") else "/votes"
-                        )
+                    if "/gov/proposals.js" in request_target_line:
```
Comment (achamayou) [inline tests/governance_history.py:49]:
```suggestion
                    if "/gov/proposals" in request_target_line:
```

File: tests/memberclient.py [hunk 8]
```diff
@@ -223,13 +220,10 @@ def test_governance(network, args):
...
 
-    if os.getenv("JS_GOVERNANCE"):
-        with primary.client(network.consortium.get_any_active_member().local_id) as c:
-            r = c.get(f"/gov/proposals.js/{proposal.proposal_id}")
-            assert r.status_code == 200, r.body.text()
-            assert (
-                r.body.json()["state"] == infra.proposal.ProposalState.WITHDRAWN.value
-            )
+    with primary.client(network.consortium.get_any_active_member().local_id) as c:
+        r = c.get(f"/gov/proposals.js/{proposal.proposal_id}")
```
Comment (achamayou) [inline tests/memberclient.py:224]:
```suggestion
        r = c.get(f"/gov/proposals/{proposal.proposal_id}")
```

File: tests/memberclient.py [hunk 9]
```diff
@@ -148,11 +146,10 @@ def test_governance(network, args):
     )
 
     LOG.info("Check proposal has been recorded in open state")
-    if os.getenv("JS_GOVERNANCE"):
-        with primary.client(network.consortium.get_any_active_member().local_id) as c:
-            r = c.get(f"/gov/proposals.js/{new_member_proposal.proposal_id}")
-            assert r.status_code == 200, r.body.text()
-            assert r.body.json()["state"] == infra.proposal.ProposalState.OPEN.value
+    with primary.client(network.consortium.get_any_active_member().local_id) as c:
+        r = c.get(f"/gov/proposals.js/{new_member_proposal.proposal_id}")
```
Comment (achamayou) [inline tests/memberclient.py:150]:
```suggestion
        r = c.get(f"/gov/proposals/{new_member_proposal.proposal_id}")
```

File: tests/memberclient.py [hunk 10]
```diff
@@ -101,7 +100,7 @@ def test_corrupted_signature(network, args):
             # Override the auth provider with invalid ones
             for fn in (missing_signature, empty_signature, modified_signature):
                 ccf.clients.RequestClient._auth_provider = make_signature_corrupter(fn)
-                r = mc.post("/gov/proposals")
+                r = mc.post("/gov/proposals.js")
```
Comment (achamayou) [inline tests/memberclient.py:103]:
```suggestion
                r = mc.post("/gov/proposals")
```

File: tests/memberclient.py [hunk 11]
```diff
@@ -20,7 +19,7 @@ def test_missing_signature_header(network, args):
     node = network.find_node_by_role()
     member = network.consortium.get_any_active_member()
     with node.client(member.local_id) as mc:
-        r = mc.post("/gov/proposals")
+        r = mc.post("/gov/proposals.js")
```
Comment (achamayou) [inline tests/memberclient.py:22]:
```suggestion
        r = mc.post("/gov/proposals")
```

File: tests/infra/member.py [hunk 12]
```diff
@@ -146,15 +143,15 @@ def propose(self, remote_node, proposal):
...
+                f"/gov/proposals.js/{proposal.proposal_id}/ballots",
                 body=ballot,
             )
 
         return r
 
     def withdraw(self, remote_node, proposal):
         with remote_node.client(*self.auth(write=True)) as c:
-            r = c.post(f"{PROPOSAL_ENDPOINT}/{proposal.proposal_id}/withdraw")
+            r = c.post(f"/gov/proposals.js/{proposal.proposal_id}/withdraw")
```
Comment (achamayou) [inline tests/infra/member.py:154]:
```suggestion
            r = c.post(f"/gov/proposals/{proposal.proposal_id}/withdraw")
```

File: tests/infra/member.py [hunk 13]
```diff
@@ -146,15 +143,15 @@ def propose(self, remote_node, proposal):
     def vote(self, remote_node, proposal, ballot):
         with remote_node.client(*self.auth(write=True)) as mc:
             r = mc.post(
-                f"{PROPOSAL_ENDPOINT}/{proposal.proposal_id}/{'ballots' if os.getenv('JS_GOVERNANCE') else 'votes'}",
+                f"/gov/proposals.js/{proposal.proposal_id}/ballots",
```
Comment (achamayou) [inline tests/infra/member.py:146]:
```suggestion
                f"/gov/proposals/{proposal.proposal_id}/ballots",
```

File: tests/infra/member.py [hunk 14]
```diff
@@ -131,7 +128,7 @@ def set_retired(self):
 
     def propose(self, remote_node, proposal):
         with remote_node.client(*self.auth(write=True)) as mc:
-            r = mc.post(f"{PROPOSAL_ENDPOINT}", proposal)
+            r = mc.post("/gov/proposals.js", proposal)
```
Comment (achamayou) [inline tests/infra/member.py:131]:
```suggestion
            r = mc.post("/gov/proposals", proposal)
```

**Review by achamayou — APPROVED** [review]

**Comment by achamayou** [comment]
@eddyashton ignore the bits about ".js", I hadn't seen the endpoints weren't renamed, I'll do that separately after this merges.

**Pull request merged**